include/picongpu/particles/collision/binary/RelativisticBinaryCollision.hpp

```diff
@@ -0,0 +1,513 @@
+                            , densitySqCbrt1(p_densitySqCbrt1)
+                            , duplicationCorrection(1u)
+                            , potentialPartners(p_potentialPartners)
+                            , coulombLog(p_coulombLog){};
+
+                        static constexpr float_COLL c = static_cast<float_COLL>(SPEED_OF_LIGHT);
+
+
+                        /* Convert momentum from the lab frame into the COM frame.
+                         *
+                         * @param labMomentum momentum in the lab frame
+                         * @param mass particle mass
+                         * @param gamma particle Lorentz factor in the lab frame
+                         * @param gammaComs @f[ \gamma_C @f] Lorentz factor of the COM frame in the lab frame
+                         * @param factorA @f[ \frac{\gamma_C - 1}{v_C^2} @f]
+                         * @param comsVelocity @f[ v_C @f] COM system velocity in the lab frame
+                         */
+                        static DINLINE float3_COLL labToComs(
+                            float3_COLL labMomentum,
+                            float_COLL mass,
+                            float_COLL gamma,
+                            float_COLL gammaComs,
+                            float_COLL factorA,
+                            float3_COLL comsVelocity)
+                        {
+                            float3_COLL labVelocity = labMomentum / gamma / mass;
+                            float_COLL dot = pmacc::math::dot(comsVelocity, labVelocity);
+                            float_COLL factor = (factorA * dot - gammaComs);
+                            factor *= mass * gamma;
+                            float3_COLL diff = factor * comsVelocity;
+                            return labMomentum + diff;
+                        }
+
+                        /* Convert momentum from the COM frame into the lab frame.
+                         *
+                         * @param labMomentum momentum in the COM frame
+                         * @param mass particle mass
+                         * @param gamma particle Lorentz factor in the lab frame
+                         * @param gammaComs @f[ \gamma_C @f] Lorentz factor of the COM frame in the lab frame
+                         * @param factorA @f[ \frac{\gamma_C - 1}{v_C^2} @f]
+                         * @param comsVelocity @f[ v_C @f] COM system velocity in the lab frame
+                         */
+                        static DINLINE float3_COLL comsToLab(
+                            float3_COLL comsMomentum,
+                            float_COLL mass,
+                            float_COLL coeff,
+                            float_COLL gammaComs,
+                            float_COLL factorA,
+                            float3_COLL comsVelocity)
+                        {
+                            // (13) in [Perez 2012]
+                            // TODO: try upgrading this to 64bit as well
```

Comment:
Isn't it already 64 bit, since `float_COLL` is `float_64`, or do you mean that is should stay 64 bit even if `float_COLL` becomes `float_32`?